RecoLocalCalo/EcalRecAlgos/src/EcalUncalibRecHitMultiFitAlgo_gpu.cu

```diff
@@ -0,0 +1,1033 @@
+        sizeof(float) * h_data.time_bias_corrections->EBTimeCorrAmplitudeBins.size(),
+        cudaMemcpyHostToDevice);
+    cudaMemcpy(d_data.EBTimeCorrShiftBins, 
+        h_data.time_bias_corrections->EBTimeCorrShiftBins.data(),
+        sizeof(float) * h_data.time_bias_corrections->EBTimeCorrShiftBins.size(),
+        cudaMemcpyHostToDevice);
+    cudaMemcpy(d_data.EETimeCorrAmplitudeBins, 
+        h_data.time_bias_corrections->EETimeCorrAmplitudeBins.data(),
+        sizeof(float) * h_data.time_bias_corrections->EETimeCorrAmplitudeBins.size(),
+        cudaMemcpyHostToDevice);
+    cudaMemcpy(d_data.EETimeCorrShiftBins, 
+        h_data.time_bias_corrections->EETimeCorrShiftBins.data(),
+        sizeof(float) * h_data.time_bias_corrections->EETimeCorrShiftBins.size(),
+        cudaMemcpyHostToDevice);
+    ecal::cuda::assert_if_error();
+
+#ifdef DEBUG
+    unsigned int idx = 0;
+        uint16_t const* p_current_digi = &digis_data[idx*EcalDataFrame::MAXSAMPLES];
+        DetId  current_id{ids[idx]};
+        EcalDataFrame edf{edm::DataFrame{ids[idx], p_current_digi, EcalDataFrame::MAXSAMPLES}};
+        auto const* aped = &vpedestals[idx];
+        auto const* aGain = &vgains[idx];
+        auto const* gid = &vxtals[idx];
+        auto const* aPulse = &vpulses[idx];
+        auto const* aPulseCov = &vcovariances[idx];
+
+        if (idx == 0) {
+            printf("*******************\n");
+            printf("cpu debug tid = %i\n", idx);
+            for(unsigned int iSample = 0; 
+                iSample < EcalDataFrame::MAXSAMPLES; iSample++)
+                printf("tid = %i i = %d adc = %d\n", idx, iSample, 
+                   edf.sample(iSample).adc());
+            for (int i=0; i<EcalPulseShape::TEMPLATESAMPLES; ++i)
+                printf("pulseshape[%d] = %f\n", i, aPulse->pdfval[i]);
+        }
+#endif
+    
+    //
+    // launch 
+    // TODO: further ntreads/nblocks optimizations...
+    //
+#ifdef DEBUG
+    std::cout << "ecal::multifit::scatter()" << std::endl;
+#endif
+    int nthreads_per_block = 256;
+    int nblocks = (h_data.digis->size() + nthreads_per_block - 1) / nthreads_per_block;
+    kernel_reconstruct<<<nblocks, nthreads_per_block>>>(
+        d_data.digis_data,
+        d_data.ids,
+        /* d_rechits, */
+        d_data.pedestals,
+        d_data.gains,
+        d_data.xtals,
+        d_data.pulses,
+        d_data.covariances,
+        d_data.rechits,
+        d_data.noisecors,
+        d_data.sample_mask,
+        d_data.EBTimeCorrAmplitudeBins, 
+        h_data.time_bias_corrections->EBTimeCorrAmplitudeBins.size(),
+        d_data.EBTimeCorrShiftBins, 
+        h_data.time_bias_corrections->EBTimeCorrShiftBins.size(),
+        d_data.EETimeCorrAmplitudeBins, 
+        h_data.time_bias_corrections->EETimeCorrAmplitudeBins.size(),
+        d_data.EETimeCorrShiftBins, 
+        h_data.time_bias_corrections->EETimeCorrShiftBins.size(),
+        h_data.digis->size()
+    );
+    cudaDeviceSynchronize();
```

Comment:
Eventually needs to be removed and the synchronization point (before the copy-back-to-host) replaced with `ExternalWork`.